frontend/src/pages/IstioConfigNew/K8sReferenceGrantForm.tsx

```diff
@@ -0,0 +1,154 @@
+import { Namespace } from '../../types/Namespace';
+import { KialiAppState } from '../../store/Store';
+import { namespaceItemsSelector } from '../../store/Selectors';
+import { KialiDispatch } from '../../types/Redux';
+import { NamespaceThunkActions } from '../../actions/NamespaceThunkActions';
+import { connect } from 'react-redux';
+
+export const K8SREFERENCEGRANT = 'K8sReferenceGrant';
+export const K8SREFERENCEGRANTS = 'k8sreferencegrants';
+
+export const FROM_KINDS = {
+  HTTPRoute: 'gateway.networking.k8s.io',
+  Gateway: 'gateway.networking.k8s.io'
+};
+
+export const TO_KINDS = {
+  Service: '',
+  Secret: ''
+};
+
+type ReduxProps = {
+  namespaces: Namespace[];
+};
+
+type ReduxDispatchProps = {
+  refresh: () => void;
+};
+
+type Props = ReduxProps &
+  ReduxDispatchProps & {
+    k8sReferenceGrant: K8sReferenceGrantState;
+    onChange: (k8sReferenceGrant: K8sReferenceGrantState) => void;
+  };
+
+export type K8sReferenceGrantState = {
+  from: K8sReferenceRule[];
+  to: K8sReferenceRule[];
+};
+
+export const initK8sReferenceGrant = (): K8sReferenceGrantState => ({
+  from: [
+    {
+      kind: Object.keys(FROM_KINDS)[0],
+      group: Object.values(FROM_KINDS)[0],
+      namespace: ''
+    }
+  ],
+  to: [{ kind: Object.keys(TO_KINDS)[0], group: Object.values(TO_KINDS)[0] }]
+});
+
+export const isK8sReferenceGrantStateValid = (g: K8sReferenceGrantState): boolean => {
+  return g.from.length > 0 && g.to.length > 0;
+};
+
+export class K8sReferenceGrantFormComponent extends React.Component<Props, K8sReferenceGrantState> {
+  constructor(props: Props) {
+    super(props);
+    this.state = initK8sReferenceGrant();
+  }
+
+  componentDidMount(): void {
+    this.props.k8sReferenceGrant.from[0].namespace = this.props.namespaces[0].name;
+    this.setState(this.props.k8sReferenceGrant);
+  }
+
+  onChangeReferenceGrantFromKind = (_event: React.FormEvent, value: string): void => {
+    this.setState(
+      {
+        from: [{ group: FROM_KINDS[value], kind: value, namespace: this.state.from[0].namespace }]
+      },
+      () => this.props.onChange(this.state)
+    );
+  };
+
+  onChangeReferenceGrantFromNamespace = (_event: React.FormEvent, value: string): void => {
+    this.setState(
+      {
+        from: [{ group: this.state.from[0].group, kind: this.state.from[0].kind, namespace: value }]
+      },
+      () => this.props.onChange(this.state)
+    );
+  };
+
+  onChangeReferenceGrantToKind = (_event: React.FormEvent, value: string): void => {
+    this.setState(
+      {
+        to: [{ group: TO_KINDS[value], kind: value }]
+      },
+      () => this.props.onChange(this.state)
+    );
+  };
+
+  render(): React.ReactNode {
+    return (
+      <>
+        <FormGroup label="From Namespace" fieldId="FromNamespace">
+          <FormSelect
+            value={this.state.from[0].namespace}
+            onChange={this.onChangeReferenceGrantFromNamespace}
+            id="ReferenceGrantClass"
```

Comment:
Why id and name value is `ReferenceGrantClass` and not `ReferenceGrantFromNamespace`? I think it this id and name represents better the select